muffnn/fm/tests/test_fm_regressor.py

```diff
@@ -0,0 +1,216 @@
+"""
+Tests for the FM Regressor
+
+based in part on sklearn's logistic tests:
+https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/linear_model/tests/test_logistic.py
+"""
+from __future__ import print_function
+from __future__ import division
+
+from io import BytesIO
+import pickle
+
+import numpy as np
+import pytest
+import scipy.sparse as sp
+from scipy.stats import pearsonr
+from sklearn.datasets import load_diabetes
+from sklearn.utils.estimator_checks import check_estimator
+from sklearn.utils.testing import assert_array_almost_equal, assert_equal
+from sklearn.preprocessing import StandardScaler
+from sklearn.model_selection import cross_val_predict, KFold
+
+from ..fm_regressor import FMRegressor
+
+# toy dataset where Y = x[0] -2 * x[1] + 2 + err
+diabetes = load_diabetes()
+X = np.array([[-1, 0], [-2, 1], [1, 1], [2, 0], [-2, 0], [0, 2]],
+             dtype=np.float32)
+Xsp = sp.csr_matrix(X)
+Y = X[:, 0] - 2 * X[:, 1] + 2 + \
+    np.random.RandomState(42).randn(X.shape[0]) * 0.01
+
+# The defaults kwargs don't work for tiny datasets like those in these tests.
+KWARGS = {"random_state": 2, "solver": 'L-BFGS-B', "rank": 1}
+SGD_KWARGS = {"random_state": 2,
+              "rank": 1,
+              "solver": 'L-BFGS-B',
+              "n_epochs": 1000}
+
+
+def check_predictions(est, X, y):
+    """Check that the model is able to fit the regression training data.
+
+    based on
+    https://github.com/scikit-learn/scikit-learn/blob/af171b84bd3fb82eed4569aa0d1f976264ffae84/sklearn/linear_model/tests/test_logistic.py#L38
+    """
+    n_samples = len(y)
+    preds = est.fit(X, y).predict(X)
+    assert_equal(preds.shape, (n_samples,))
+    assert_array_almost_equal(preds, y, decimal=1)
+
+
+# TODO remove this -- it is only used for `check_estimator`
```

Comment:
What? Why does this need to be removed?

Reply:
The reason I wrote this is that the tests were failing or too slow with the normal settings for the classifier. The fact that it is used once is NDB.